### ISSUE TITLE
Build Failure of commit 72df5af under MacOS 10.12.6  with XCode  9.0 (9A235)

### ISSUE THREAD

**Issue by pickn6** [google/shaka-packager#285]
## System info
**Operating System**: <e.g. macOS Sierra, 10.12.6>
**Shaka Packager Version**: <72df5af>

## Issue and steps to reproduce the problem
1. Follow the build instruction to get code
gclient config https://www.github.com/google/shaka-packager.git --name=src --unmanaged
gclient sync 

2. Build code
cd src
ninja -C out/Release

**Packager Command**:
N/A

What happens instead?
Build failed with the following dump: 
ninja: Entering directory `out/Release'
[158/764] CC obj/third_party/curl/source/lib/libcurl.connect.o
FAILED: obj/third_party/curl/source/lib/libcurl.connect.o 
../../packager/third_party/llvm-build/Release+Asserts/bin/clang -MMD -MF obj/third_party/curl/source/lib/libcurl.connect.o.d -DV8_DEPRECATION_WARNINGS -D__ASSERT_MACROS_DEFINE_VERSIONS_WITHOUT_UNDERSCORE=0 -DCHROMIUM_BUILD -DCR_CLANG_REVISION=274369-1 -DUSE_LIBJPEG_TURBO=1 -DENABLE_WEBRTC=1 -DENABLE_MEDIA_ROUTER=1 -DENABLE_PEPPER_CDMS -DENABLE_NOTIFICATIONS -DUSE_EXTERNAL_POPUP_MENU -DFIELDTRIAL_TESTING_ENABLED -DENABLE_TASK_MANAGER=1 -DENABLE_EXTENSIONS=1 -DENABLE_PDF=1 -DENABLE_PLUGIN_INSTALLATION=1 -DENABLE_PLUGINS=1 -DENABLE_SESSION_SERVICE=1 -DENABLE_THEMES=1 -DENABLE_PRINTING=1 -DENABLE_BASIC_PRINTING=1 -DENABLE_PRINT_PREVIEW=1 -DENABLE_SPELLCHECK=1 -DUSE_BROWSER_SPELLCHECKER=1 -DENABLE_CAPTIVE_PORTAL_DETECTION=1 -DENABLE_SUPERVISED_USERS=1 -DENABLE_SERVICE_DISCOVERY=1 -DV8_USE_EXTERNAL_STARTUP_DATA -DFULL_SAFE_BROWSING -DSAFE_BROWSING_CSD -DSAFE_BROWSING_DB_LOCAL -DCURL_STATICLIB -DHTTP_ONLY -DUSE_IPV6 -DHAVE_CONFIG_H '-DCURL_CA_BUNDLE="/usr/local/etc/openssl/cert.pem"' -DUSE_OPENSSL -DUSE_LIBPCI=1 -DNDEBUG -DNVALGRIND -DDYNAMIC_ANNOTATIONS_ENABLED=0 -Igen -I../../packager/third_party/curl/source/lib -I../../packager/third_party/curl/source/include -I../../packager/third_party/boringssl/src/include -I../../packager/third_party/curl/config -I../../packager/third_party/curl/config/curl -I../../packager/third_party/curl/config/mac -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk -O2 -gdwarf-2 -fvisibility=hidden -Werror -mmacosx-version-min=10.7 -arch x86_64 -Wall -Wno-unused-parameter -Wno-missing-field-initializers -Wno-selector-type-mismatch -Wpartial-availability -Wheader-hygiene -Wno-char-subscripts -Wno-unneeded-internal-declaration -Wno-covered-switch-default -Wstring-conversion -Wno-c++11-narrowing -Wno-deprecated-register -Wno-inconsistent-missing-override -Wno-shift-negative-value -Wno-undefined-var-template -Wno-nonportable-include-path -Wno-varargs -Wno-unused-variable -std=c99 -fcolor-diagnostics -fno-strict-aliasing  -c ../../packager/third_party/curl/source/lib/connect.c -o obj/third_party/curl/source/lib/libcurl.connect.o
../../packager/third_party/curl/source/lib/connect.c:1083:12: error: 'connectx' is partial: introduced in macOS 10.11 [-Werror,-Wpartial-availability]
      rc = connectx(sockfd, &endpoints, SAE_ASSOCID_ANY,
           ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk/usr/include/sys/socket.h:713:5: note: 'connectx' has been explicitly marked partial here
int connectx(int, const sa_endpoints_t *, sae_associd_t, unsigned int,
    ^
../../packager/third_party/curl/source/lib/connect.c:1083:12: note: explicitly redeclare 'connectx' to silence this warning
      rc = connectx(sockfd, &endpoints, SAE_ASSOCID_ANY,
           ^
1 error generated.
[167/764] CXX obj/media/formats/mp4/mp4.box_definitions.o
ninja: build stopped: subcommand failed.

<Please attach the input files or email to shaka-packager-issues@google.com.>
<Please provide any additional information below.>

**Comment by kqyang** [comment]
Thanks for reporting the issue. Can you confirm the OS version is macOS Sierra, 10.12.6? I tried the same exact version, but I am not seeing problems.

**Comment by pickn6** [comment]
MacOS 10.12.6 (16G29), XCode 9.0 (9A235). 

The symptom is similar to that in https://github.com/JuliaLang/julia/issues/23980

**Comment by kqyang** [comment]
Thanks for the information! So the problem is likely related to XCode 9.0. The same fix in JuliaLang/julia#23980 should work.

Do you mind verify if it could fix the problem? Steps:

1. Update [DEPS](https://github.com/google/shaka-packager/blob/master/DEPS#L42) file

Replace 

    Var("github") + "/curl/curl@79e63a53bb9598af863b0afe49ad662795faeef4",  #7.50.0

With

    Var("github") + "/curl/curl@c514af5a4f5ac3ce724065cc6a8e009373436f78",  #7.56.1

2. Run ```gclient sync```

3. Try compile again

**Comment by pickn6** [comment]
Is there other files that needs to be updated as well? If only curl source updated, there will be an compiling error: 
ninja: error: '../../packager/third_party/curl/source/lib/rawstr.c', needed by 'obj/third_party/curl/source/lib/libcurl.rawstr.o', missing and no known rule to make it

**Comment by kqyang** [comment]
Emm... Looks like some files have been removed. Not sure if `rawstr.c` is the only file removed.

https://github.com/google/shaka-packager/blob/master/packager/third_party/curl/curl.gyp needs to be updated. `rawstr.c` should be removed from the list.

After the change, you'll need to run `gclient runhooks` for gyp change to take effect.

Thanks very much for your help! (It is up to you if you want to proceed. We can take it over from there. We'll just need to find a setup with XCode 9.0).

**Comment by kqyang** [comment]
@pickn6 The issue should be fixed with curl roll in 9631ec7. Please re-open the bug if it is not. Thanks.

**Comment by aaani** [comment]
@kqyang I am still noticing this issue on the latest master
OS: Sierra 10.12.6
Shaka Packager last commit: 66f713fd23

```
➜  src git:(66f713fd23) ninja -C out/Release
ninja: Entering directory `out/Release'
[297/1542] CC obj/third_party/curl/source/lib/libcurl.connect.o
FAILED: obj/third_party/curl/source/lib/libcurl.connect.o 
../../packager/third_party/llvm-build/Release+Asserts/bin/clang -MMD -MF obj/third_party/curl/source/lib/libcurl.connect.o.d -DV8_DEPRECATION_WARNINGS -D__ASSERT_MACROS_DEFINE_VERSIONS_WITHOUT_UNDERSCORE=0 -DCHROMIUM_BUILD -DCR_CLANG_REVISION=274369-1 -DUSE_LIBJPEG_TURBO=1 -DENABLE_WEBRTC=1 -DENABLE_MEDIA_ROUTER=1 -DENABLE_PEPPER_CDMS -DENABLE_NOTIFICATIONS -DUSE_EXTERNAL_POPUP_MENU -DFIELDTRIAL_TESTING_ENABLED -DENABLE_TASK_MANAGER=1 -DENABLE_EXTENSIONS=1 -DENABLE_PDF=1 -DENABLE_PLUGIN_INSTALLATION=1 -DENABLE_PLUGINS=1 -DENABLE_SESSION_SERVICE=1 -DENABLE_THEMES=1 -DENABLE_PRINTING=1 -DENABLE_BASIC_PRINTING=1 -DENABLE_PRINT_PREVIEW=1 -DENABLE_SPELLCHECK=1 -DUSE_BROWSER_SPELLCHECKER=1 -DENABLE_CAPTIVE_PORTAL_DETECTION=1 -DENABLE_SUPERVISED_USERS=1 -DENABLE_SERVICE_DISCOVERY=1 -DV8_USE_EXTERNAL_STARTUP_DATA -DFULL_SAFE_BROWSING -DSAFE_BROWSING_CSD -DSAFE_BROWSING_DB_LOCAL -DBUILDING_LIBCURL -DCURL_STATICLIB -DHTTP_ONLY -DUSE_IPV6 -DHAVE_CONFIG_H '-DCURL_CA_BUNDLE="/usr/local/etc/openssl/cert.pem"' -DUSE_OPENSSL -DUSE_LIBPCI=1 -DNDEBUG -DNVALGRIND -DDYNAMIC_ANNOTATIONS_ENABLED=0 -Igen -I../../packager/third_party/curl/source/lib -I../../packager/third_party/curl/source/include -I../../packager/third_party/boringssl/src/include -I../../packager/third_party/curl/config -I../../packager/third_party/curl/config/curl -I../../packager/third_party/curl/config/mac -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk -O2 -gdwarf-2 -fvisibility=hidden -Werror -mmacosx-version-min=10.7 -arch x86_64 -Wall -Wno-unused-parameter -Wno-missing-field-initializers -Wno-selector-type-mismatch -Wpartial-availability -Wheader-hygiene -Wno-char-subscripts -Wno-unneeded-internal-declaration -Wno-covered-switch-default -Wstring-conversion -Wno-c++11-narrowing -Wno-deprecated-register -Wno-inconsistent-missing-override -Wno-shift-negative-value -Wno-undefined-var-template -Wno-nonportable-include-path -Wno-varargs -Wno-unused-variable -std=c99 -fcolor-diagnostics -fno-strict-aliasing  -c ../../packager/third_party/curl/source/lib/connect.c -o obj/third_party/curl/source/lib/libcurl.connect.o
../../packager/third_party/curl/source/lib/connect.c:1072:14: error: 'connectx' is partial: introduced in macOS 10.11 [-Werror,-Wpartial-availability]
        rc = connectx(sockfd, &endpoints, SAE_ASSOCID_ANY,
             ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk/usr/include/sys/socket.h:713:5: note: 'connectx' has been explicitly marked partial here
int connectx(int, const sa_endpoints_t *, sae_associd_t, unsigned int,
    ^
../../packager/third_party/curl/source/lib/connect.c:1072:14: note: explicitly redeclare 'connectx' to silence this warning
        rc = connectx(sockfd, &endpoints, SAE_ASSOCID_ANY,
             ^
1 error generated.
[306/1542] CC obj/third_party/boringssl/src/crypto/fipsmodule/boringssl.bcm.o
ninja: build stopped: subcommand failed.
```

**Comment by kqyang** [comment]
It turns out the problem is fixed in curl 7.57.0 instead of 7.56.0.

@aaani Can you try again with commit 5f48cbb? It should be fixed this time. Let us know if it is not.
